concent_api/middleman/middleman_server.py

```diff
@@ -108,26 +115,43 @@ def _close_middleman(self) -> None:
 
     async def _handle_concent_connection(self, reader, writer) -> None:
         try:
-            data = await reader.readuntil()
+            data = await handle_frame_receive_async(reader)
             remote_address = writer.get_extra_info('peername')
             print("Received %r from %r" % (data, remote_address))
-            await self._respond_to_user(data, writer)
+            serialized_message = self._prepare_response(data)
+            await self._respond_to_user(serialized_message, writer)
         except Exception as exception:  # pylint: disable=broad-except
             crash_logger.error(
                 f"Exception occurred: {exception}, Traceback: {traceback.format_exc()}"
             )
             raise
 
+    @staticmethod
+    def _prepare_response(data):
+        try:
+            AbstractFrame.deserialize(data, settings.CONCENT_PUBLIC_KEY)
+            serialized_message = data
+        except MiddlemanProtocolError as exception:
+            error_code = map_exception_to_error_code(exception)
+            serialized_message = ErrorFrame(
+                (error_code, str(exception)),
+                -1,
```

Comment:
This value will not work with current implementation of protocol, but this can be easily changed in protocol (`request_id` should be `BytesInteger(signed=True)`), but be careful as probably `FRAME_REQUEST_ID_BYTES_LENGTH` should be increased by 1 then.

Reply:
chaqnged to `0`

Reply:
done

Reply:
I have also changed this in `SigningService`, but I am thinking if this should be a part of procotol, what do you guys think (I mean nicely named constant equal to `0`, and maybe validation to make sure `ErrorFrame` will always use it) ?